File: demo-routines/src/main/java/at/enfilo/def/demo/PICalc.java
package at.enfilo.def.demo;

import at.enfilo.def.datatype.DEFDouble;
import at.enfilo.def.routine.AccessParameterException;
import at.enfilo.def.routine.ObjectiveRoutine;
import at.enfilo.def.routine.RoutineException;

public class PICalc extends ObjectiveRoutine<DEFDouble> {

	@Override
	protected DEFDouble routine() throws RoutineException {
		try {
			double start = getParameter("start", DEFDouble.class).getValue();
			double end = getParameter("end", DEFDouble.class).getValue();
			double stepSize = getParameter("stepSize", DEFDouble.class).getValue();

			double sum = 0.0;
			for (double i = start; i < end; i++) {
				double x = (i + 0.5) * stepSize;
				sum += 4.0 / (1.0 + x * x);
			}
			sum *= stepSize;

			System.out.println("Pi part successful calculated: " + sum);

			return new DEFDouble(sum);

		} catch (AccessParameterException e) {
			throw new RoutineException(e);
		}
	}
}

File: apis/csharp/demo-routines/PiCalc.cs
using routine_api;
using routine_api.exception;

namespace demo_routines
{
    public class PiCalc : ObjectiveRoutine<DEFDouble>
    {
        protected override DEFDouble Routine()
        {
            try
            {
                double start = GetParameter<DEFDouble>("start").Value;
                double end = GetParameter<DEFDouble>("end").Value;
                double stepSize = GetParameter<DEFDouble>("stepSize").Value;

                double sum = 0.0;
                for (double i = start; i < end; i++)
                {
                    double x = (i + 0.5) * stepSize;
                    sum += 4.0 / (1.0 + x * x);
                }

                sum *= stepSize;

                return new DEFDouble
                {
                    Value = sum
                };
            }
            catch (AccessParameterException e)
            {
                throw new RoutineException(e);
            }
        }
    }
}